tsl/src/continuous_aggs/create.c

```diff
@@ -1097,38 +1110,63 @@ cagg_validate_query(Query *query, bool finalized)
 	/* check if we have a hypertable in the FROM clause */
 	rtref = linitial_node(RangeTblRef, query->jointree->fromlist);
 	rte = list_nth(query->rtable, rtref->rtindex - 1);
+
 	/* FROM only <tablename> sets rte->inh to false */
-	if (rte->relkind != RELKIND_RELATION || rte->tablesample || rte->inh == false)
+	if ((rte->relkind != RELKIND_RELATION && rte->relkind != RELKIND_VIEW) || rte->tablesample ||
+		rte->inh == false)
 	{
 		ereport(ERROR,
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("invalid continuous aggregate view")));
 	}
-	if (rte->relkind == RELKIND_RELATION)
+
+	if (rte->relkind == RELKIND_RELATION || rte->relkind == RELKIND_VIEW)
 	{
 		const Dimension *part_dimension = NULL;
+		int32 parent_mat_hypertable_id = INVALID_HYPERTABLE_ID;
+
+		if (rte->relkind == RELKIND_RELATION)
+			ht = ts_hypertable_cache_get_cache_and_entry(rte->relid, CACHE_FLAG_NONE, &hcache);
+		else
+		{
+			const ContinuousAgg *cagg;
 
-		ht = ts_hypertable_cache_get_cache_and_entry(rte->relid, CACHE_FLAG_NONE, &hcache);
+			cagg = ts_continuous_agg_find_by_relid(rte->relid);
+
+			if (!ContinuousAggIsFinalized(cagg))
+			{
+				ereport(ERROR,
+						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						 errmsg("old format of continuous aggregate is not supported for this "
+								"feature"),
+						 errhint("Run \"CALL cagg_migrate('%s.%s');\" to migrate to the new "
+								 "format.",
+								 NameStr(cagg->data.user_view_schema),
+								 NameStr(cagg->data.user_view_name))));
+			}
+
+			parent_mat_hypertable_id = cagg->data.mat_hypertable_id;
+			hcache = ts_hypertable_cache_pin();
+			ht = ts_hypertable_cache_get_entry_by_id(hcache, cagg->data.mat_hypertable_id);
```

Comment:
Don't you need a check that the cagg's time_bucket interval is a multiple of the interval of the original cagg.
i.e. if the first cagg has time_bucket 15 minutes,
cagg on cagg cannot have time_bucket of 20 minutes

Reply:
You're correct... will take a note in this PR about it and another validations as we discussed in PVT.